### PR TITLE
UI: Add audio mixer toolbar

### PR DESCRIPTION
**Pull request by cg2121**
### Description
This adds a toolbar to the mixer dock, similar to scenes
and sources. This also moves the vertical toggle and
unhide all action to this toolbar.

mixer-toolbar

### Motivation and Context
This makes it easier for users to find the advanced audio properties dialog.

### How Has This Been Tested?
Opened OBS to make sure the button looked correct and clicked it to make sure the advanced audio dialog opened.

### Types of changes
- New feature (non-breaking change which adds functionality)

### Checklist:
- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

File: UI/forms/OBSBasic.ui [hunk 1]
```diff
@@ -961,7 +961,13 @@
    <attribute name="dockWidgetArea">
     <number>8</number>
    </attribute>
-   <widget class="QWidget" name="dockWidgetContents_7">
```
Comment (Warchamp7) [inline UI/forms/OBSBasic.ui:964]: This should be a QFrame within a QWidget similar to the Scenes and Sources dock

image

**Comment by Warchamp7** [comment]
Left a note on the .ui file semantics, but this is a good change on the UX side.

File: UI/forms/OBSBasic.ui (hunk 1 above)
Comment (Warchamp7) [reply UI/forms/OBSBasic.ui:964]: This is out of scope for this PR so I'm retracting it

**Comment by WizardCM** [comment]
I would also like to see us potentially move more things out of the context menu and into this toolbar, such as the horizontal/vertical toggle.

**Comment by Warchamp7** [comment]
Sorry I meant to move these options into a menu under the gear, not to add them as toolbar buttons.

I'm not sure either of these actions can be communicated well with an icon.

**Comment by cg2121** [comment]
Updated to put the unhide all and toggle behind menu button

Screenshot from 2022-07-25 08-05-31

**Comment by Warchamp7** [comment]
The gear here opens Advanced Audio Properties right?

Can we use the multi gear icon instead (Used for the Settings -> Advanced tab currently)

**Comment by cg2121** [comment]
Updated to use advanced icon

**Comment by cg2121** [comment]
Updated the advanced audio icon to be a 2 cogs icon
Icon is from https://github.com/fiction-com/ficons

mixer-toolbar

**Comment by cg2121** [comment]
Closed. It is now combined with #6756 because they cause merge conflicts with each other.

**Comment by cg2121** [comment]
Reopening, because it was decided off thread that the transition toolbar was not the right design, so only add the mixer toolbar for now.

**Pull request merged**